File: from-1701-to-1800/1750-minimum-length-of-string-after-deleting-similar-ends/solution.py
class Solution:
    def minimumLength(self, s: str) -> int:
        if len(s) == 1:
            return 1
        
        left = 0
        right = len(s) - 1
        
        while left < right and s[left] == s[right]:
            while s[left] == s[left+1] and left+1 < right:
                left += 1
            left += 1
            while s[right] == s[right-1] and right - 1 > left:
                right -= 1
            right -= 1
                
        return right - left + 1
        

File: from-1701-to-1800/1750-minimum-length-of-string-after-deleting-similar-ends/solution.go
func minimumLength(s string) int {
    left, right := 0, len(s) - 1
    for left < right && s[left] == s[right] {
        for s[left] == s[left + 1] && left + 1 < right {
            left += 1
        }
        left += 1

        for s[right - 1] == s[right] && right - 1 > left {
            right -= 1
        }
        right -= 1
    }
    return right - left + 1
}